packages/flutter/lib/src/widgets/scroll_position_with_single_context.dart

```diff
@@ -143,12 +144,37 @@ class ScrollPositionWithSingleContext extends ScrollPosition implements ScrollAc
     assert(hasPixels);
     final Simulation? simulation = physics.createBallisticSimulation(this, velocity);
     if (simulation != null) {
-      beginActivity(BallisticScrollActivity(this, simulation, context.vsync));
+      beginActivity(BallisticScrollActivity(
+        this,
+        simulation,
+        context.vsync,
+        initVelocity: velocity,
+        initPosition: pixels,
+      ));
     } else {
       goIdle();
     }
   }
 
+  @override
+  Simulation? updateBallisticAnimation(double initVelocity, double initPosition) {
+    assert(hasPixels);
+    final FixedScrollMetrics initScrollMetrics = FixedScrollMetrics(
+      minScrollExtent: minScrollExtent,
+      maxScrollExtent: maxScrollExtent,
+      pixels: initPosition,
+      viewportDimension: viewportDimension,
+      axisDirection: axisDirection,
+    );
+    final Simulation? simulation = physics.createBallisticSimulation(initScrollMetrics, initVelocity);
+    if (simulation != null) {
+      return simulation;
+    } else {
+      goIdle();
+      return null;
+    }
```

Comment:
Maybe a little easier to read:

```suggestion
    if (simulation == null) {
      goIdle();
      return null;
    }
    return simulation;
```